R/assertions.R

```diff
@@ -132,30 +133,41 @@ assert_data_frame <- function(arg,
 assert_character_scalar <- function(arg,
                                     values = NULL,
                                     case_sensitive = TRUE,
-                                    optional = FALSE) {
+                                    optional = FALSE,
+                                    arg_name = rlang::caller_arg(arg),
+                                    message = NULL,
+                                    class = "assert_character_scalar",
+                                    call = parent.frame()) {
   assert_character_vector(values, optional = TRUE)
   assert_logical_scalar(optional)
 
   if (optional && is.null(arg)) {
     return(invisible(arg))
   }
 
+  message <-
+    message %||%
+    ifelse(
+      is.null(values),
+      "Argument {.arg {arg_name}} must be a scalar of class {.cls character},
+       but is {.obj_type_friendly {arg}}.",
+      "Argument {.arg {arg_name}} must be equal to one of {.val {values}}."
```

Comment:
should "one of" be removed when `values` contains one value
```r
> assert_character_scalar("1", "a")
Error:
! Argument `"1"` must be equal to one of "a".
Run `rlang::last_error()` to see where the error occurred.
```

Reply:
I don't think this is the type of check a developer would write to check for a single value? Is there an example of this in admiral?

Reply:
You are right - do you think it's necessary to add a length restriction >1 for the input `values`?

Reply:
Hmmm good question. I think we can leave it as it is so we don't make unnecessary changes to the current implementation of the check, rather focussing on the messaging to the users.

If we want to make this change, perhaps we can make that change in another PR that focusses on that change to the functionality?